### ISSUE TITLE
Thread dump on successive TransactionManager creation (even after closing)

### ISSUE THREAD

**Issue by jeremyk-91** [palantir/atlasdb#1234]
@j-baker for SA

I was able to reproduce an issue an internal product is facing with their use case where in the same run they create and close multiple `TransactionManager`s, though their lifespans are non-overlapping (i.e. there is at most one open at any time). This seems reasonable, yet they are encountering thread dumps in their logs.

Simple repro:
In `AtlasDbEteServer` change `run()` to:

```java
    @Override
    public void run(AtlasDbEteConfiguration config, final Environment environment) throws Exception {
        TransactionManager transactionManager = createTransactionManager(config, environment);
        transactionManager.close();
        TransactionManager transactionManager2 = createTransactionManager(config, environment);
        transactionManager2.close();
    }
```

This currently results in a thread dump being logged, which we should ideally avoid. Probably because we use a static reference in `ServiceDiscoveringAtlasSupplier` - a possible fix could be to find some way of passing a TransactionManager `close()` to nullify the reference, though I'm not sure if that has further implications.

**Comment by jeremyk-91** [comment]
Discussed with @gsheasby - this is nontrivial and as far as we know this use case doesn't occur in practice. A workaround could be to switch off the logging for the time being. This will go away with external Timelock Servers.

**Comment by gsheasby** [comment]
To make the logging less painful but still useful in the error case, we could either:
- compress the output using Base64 encoding (hacky but it won't look like a stack trace anymore!)
- print out the stack trace just for the current thread

**Comment by gsheasby** [comment]
Fixed by #1275
